src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -1237,9 +1279,12 @@ def add_units_table(
         spike_times = np.concatenate(spike_times)
         if waveform_means is not None:
             unit_kwargs["waveform_mean"] = waveform_means[row]
-        if waveform_sds is not None:
-            unit_kwargs["waveform_sd"] = waveform_sds[row]
+            if waveform_sds is not None:
+                unit_kwargs["waveform_sd"] = waveform_sds[row]
+            if unit_electrode_indices is not None:
+                unit_kwargs["electrodes"] = unit_electrode_indices[row]
         units_table.add_unit(spike_times=spike_times, **unit_kwargs, enforce_unique_id=True)
+    added_unit_table_ids = units_table.id[-len(rows_to_add) :]
```

Comment:
@CodyCBakerPhD this is needed to keep old behavior of `write_waveforms`: as now table ids are not necessary coincident with unit ids, we need to pass the list of added unit ids to the `_add_waveforms_to_units_table()` function.

Reply:
Thanks!